src/eager/setup.py

```diff
@@ -43,39 +47,56 @@ def is_debug_build():
 pytorch_compile_commands_path = os.path.join(
   pytorch_src_dir, 'compile_commands.json')
 
-if not os.path.exists(os.path.join(pytorch_src_dir, '.git')):
+if not args.use_preinstalled_torch and not os.path.exists(os.path.join(pytorch_src_dir, '.git')):
   raise Exception('pytorch submodule does not exist. Run git submodule update --init --recursive')
 
 ort_src_dir = os.path.join(repo_root_dir, 'external', 'onnxruntime')
 ort_build_dir = os.path.join(self_dir, 'ort_build', build_config)
 
-if not os.path.exists(os.path.join(ort_src_dir, '.git')):
+if not args.ort_path and not os.path.exists(os.path.join(ort_src_dir, '.git')):
   raise Exception('onnxruntime submodule does not exist. Run git submodule update --init --recursive')
 
-ort_lib_dirs = [
-  ort_build_dir
-]
+ort_lib_dirs = [ort_build_dir] if not args.ort_path else []
 
 import numpy as np
 
-ort_include_dirs = [
-  os.path.join(ort_src_dir, 'include', 'onnxruntime'),
-  os.path.join(ort_src_dir, 'include', 'onnxruntime', 'core', 'session'),
-  os.path.join(ort_src_dir, 'onnxruntime'),
-  os.path.join(ort_src_dir, 'orttraining'),
-  os.path.join(ort_build_dir),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'onnx'),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'SafeInt'),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'protobuf', 'src'),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'nsync', 'public'),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'mp11', 'include'),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'optional-lite', 'include'),
-  os.path.join(ort_src_dir, 'cmake', 'external', 'dlpack', 'include'),
-  os.path.join(ort_build_dir, 'external', 'onnx'),
-  np.get_include()
-]
+if not args.ort_path:
```

Comment:
there is some duplicated code in both if and else branches. Maybe some refactoring to have a common list of folders without conditionals and only the specific stuff wrapped on the conditional? files are added and removed all the time and one conditional can get out of sync very easy

Reply:
Refactored code to reuse common include dirs